### PR TITLE
Add support for C-language annotations.

### PR DESCRIPTION
**Pull request by jverkoey**
Example annotation:

```
- (void)someMethod __attribute__((annotate("This is a custom annotation.")));
```

Will appear in the output as part of a new `annotations` field as `["This is a custom annotation."]`.

### CODE REVIEW

**Comment by jverkoey** [comment]
For context, we hope to use annotations to indicate that APIs are "to be deprecated". This will feed in to our [API diff](https://github.com/material-motion/apidiff) toolchain in order to give early notice for APIs will be deprecated (without immediately escalating to a build-time warning using the `__deprecated` annotation).

**Comment by jverkoey** [comment]
Trying to remember how to re-generate the test outputs....

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
 ##### Enhancements
 
-* None.
+* Add support for annotations.
```
Comment (jpsim) [inline CHANGELOG.md]: Can you be more specific, like `Add support for C-language annotations.`?

Comment (jverkoey) [reply CHANGELOG.md]: Done.

**Comment by jverkoey** [comment]
I think tests should be passing now.

File: Source/SourceKittenFramework/SourceDeclaration.swift [hunk 2]
```diff
@@ -141,12 +141,17 @@ extension SourceDeclaration {
...
         availability = cursor.platformAvailability()
 
-        annotations = cursor.compactMap({
+        let annotations: [String] = cursor.compactMap({
             if $0.kind == CXCursor_AnnotateAttr {
                 return clang_getCursorSpelling($0).str()!
             }
             return nil
         })
+        if annotations.count > 0 {
```
Comment (jpsim) [inline Source/SourceKittenFramework/SourceDeclaration.swift]: isEmpty

File: Source/SourceKittenFramework/SourceDeclaration.swift [hunk 3]
```diff
@@ -139,6 +140,18 @@ extension SourceDeclaration {
...
         (swiftDeclaration, swiftName) = cursor.swiftDeclarationAndName(compilerArguments: compilerArguments)
         availability = cursor.platformAvailability()
+
+        let annotations: [String] = cursor.compactMap({
+            if $0.kind == CXCursor_AnnotateAttr {
+                return clang_getCursorSpelling($0).str()!
+            }
+            return nil
+        })
+        if annotations.count > 0 {
```
Comment (jpsim) [inline Source/SourceKittenFramework/SourceDeclaration.swift]: isEmpty

File: Source/SourceKittenFramework/SourceDeclaration.swift [hunk 4]
```diff
@@ -139,6 +140,18 @@ extension SourceDeclaration {
         }).rejectPropertyMethods()
         (swiftDeclaration, swiftName) = cursor.swiftDeclarationAndName(compilerArguments: compilerArguments)
         availability = cursor.platformAvailability()
+
+        let annotations: [String] = cursor.compactMap({
+            if $0.kind == CXCursor_AnnotateAttr {
+                return clang_getCursorSpelling($0).str()!
```
Comment (jpsim) [inline Source/SourceKittenFramework/SourceDeclaration.swift]: You can remove this force unwrap

File: Source/SourceKittenFramework/SourceDeclaration.swift (hunk 3 above)
Comment (jverkoey) [reply Source/SourceKittenFramework/SourceDeclaration.swift]: Done.

File: Source/SourceKittenFramework/SourceDeclaration.swift (hunk 4 above)
Comment (jverkoey) [reply Source/SourceKittenFramework/SourceDeclaration.swift]: Done.

**Comment by jverkoey** [comment]
CI has passed.

**Comment by jpsim** [comment]
Nice! I assume you'd like a release with this?

**Comment by jverkoey** [comment]
A release would be swell - happy to help in any way <3

**Comment by jverkoey** [comment]
Hiya @jpsim - following up on whether there's anything I can do to help cut a release?

**Comment by jpsim** [comment]
Sorry for the delay here. I started the process to make a release, I'll post here when it's done.

**Comment by jpsim** [comment]
Done: https://github.com/jpsim/SourceKitten/releases/tag/0.21.2

**Comment by jverkoey** [comment]
Thank you!

**Pull request merged**